src/libaktualizr/crypto/crypto.cc

```diff
@@ -345,3 +397,37 @@ bool Crypto::generateKeyPair(KeyType key_type, std::string *public_key, std::str
   }
   return Crypto::generateRSAKeyPair(key_type, public_key, private_key);
 }
+
+bool Crypto::IsRsaKeyType(KeyType type) {
+  switch (type) {
+    case kRSA2048:
+    case kRSA4096:
+      return true;
+    default:
+      return false;
+  }
+}
+KeyType Crypto::IdentifyRSAKeyType(const std::string &public_key_pem) {
+  StructGuard<BIO> bufio(BIO_new_mem_buf(reinterpret_cast<const void *>(public_key_pem.c_str()),
+                                         static_cast<int>(public_key_pem.length())),
+                         BIO_vfree);
+  if (bufio.get() == nullptr) {
+    throw std::runtime_error("BIO_new_mem_buf failed");
+  }
+  StructGuard<::RSA> rsa(PEM_read_bio_RSA_PUBKEY(bufio.get(), nullptr, nullptr, nullptr), RSA_free);
```

Comment:
I'm also quite skeptical that this is a good idea. This looks to me like it is begging for some sort of collision or prepared attack. It also privileges RSA in a way that seems less than advisable considering that I thought we wanted to move to ED25519.

Reply:
The difficulty is that the Uptane protocol just talks about 'RSA' keys, and we need a way to convert those into a `PublicKey`. I can see three options here:

* Autodetection (what is currently implemented): We are given `keytype="rsa", keyvalue="abc123..."`, and look at the value to see what kind of key it is.
* Introduce a new 'kRSAANY' type to KeyType, which means 'Any length of RSA key'
* Introduce a parallel enum with just ED25519 and RSA options (feels ugly)

I don't think autodetection is so bad. It is basically part of the process of decoding a Uptane formated key, where once we know it is RSA, we take a look at the bytes and figure out how long it is.